instrumentation/struts-2.3/javaagent/src/test/java/io/opentelemetry/struts/GreetingAction.java

```diff
@@ -20,25 +20,38 @@ public String success() {
     return "greeting";
   }
 
-  public String query() {
+  public String redirect() {
+    responseBody =
+        HttpServerTest.controller(
+            HttpServerTest.ServerEndpoint.REDIRECT,
+            HttpServerTest.ServerEndpoint.REDIRECT::getBody);
+    return "redirect";
+  }
+
+  public String query_param() {
     responseBody =
         HttpServerTest.controller(
             HttpServerTest.ServerEndpoint.QUERY_PARAM,
             HttpServerTest.ServerEndpoint.QUERY_PARAM::getBody);
     return "greeting";
   }
 
+  public String error() {
+    HttpServerTest.controller(
+        HttpServerTest.ServerEndpoint.ERROR, HttpServerTest.ServerEndpoint.ERROR::getBody);
+    return "error";
+  }
+
   public String exception() {
-    responseBody =
-        HttpServerTest.controller(
-            HttpServerTest.ServerEndpoint.EXCEPTION,
-            () -> {
-              throw new Exception(HttpServerTest.ServerEndpoint.EXCEPTION.getBody());
-            });
-    return "exception";
+    HttpServerTest.controller(
+        HttpServerTest.ServerEndpoint.EXCEPTION,
+        () -> {
+          throw new Exception(HttpServerTest.ServerEndpoint.EXCEPTION.getBody());
+        });
+    throw new AssertionError(); // should not reach here
   }
 
-  public String pathParam() {
+  public String path_param() {
```

Comment:
Why snake case?

Reply:
I believe this is needed for `def expectedMethodName = endpoint.name().toLowerCase()` to work in the test.

Reply:
ya, this "technique" is used in a few other tests as well, just staying consistent 😄